yarn-project/simulator/src/acvm/acvm.ts

```diff
@@ -18,7 +18,15 @@ import { type ORACLE_NAMES } from './oracle/index.js';
  */
 type ACIRCallback = Record<
   ORACLE_NAMES,
-  (...args: ForeignCallInput[]) => void | Promise<void> | ForeignCallOutput | Promise<ForeignCallOutput>
+  (
+    ...args: ForeignCallInput[]
+  ) =>
+    | void
+    | Promise<void>
+    | ForeignCallOutput
+    | ForeignCallOutput[]
+    | Promise<ForeignCallOutput>
+    | Promise<ForeignCallOutput[]>
```

Comment:
We didn't support returning of non-flat structures from the oracles so I had to do this ^